projects/aca-shared/src/lib/services/aca-file-auto-download.service.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { AppConfigService } from '@alfresco/adf-core';
+import { NodeEntry } from '@alfresco/js-api';
+import { FileAutoDownloadComponent } from '@alfresco/adf-content-services';
+
+const BYTES_TO_MB_CONVERSION_VALUE = 1048576;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AcaFileAutoDownloadService {
```

Comment:
there's already a https://github.com/Alfresco/alfresco-content-app/blob/4956655a801bdb714fa76399df0fe81d8c5976c0/projects/aca-content/src/lib/store/effects/download.effects.ts#L37 containing various effects for dowloading